src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -37,24 +38,72 @@ class Neo4jQueryWriteStrategy(private val saveMode: SaveMode) extends Neo4jQuery
   }
 }
 
-class Neo4jQueryReadStrategy extends Neo4jQueryStrategy {
+class Neo4jQueryReadStrategy(filters: Array[Filter]) extends Neo4jQueryStrategy {
   private val renderer: Renderer = Renderer.getDefaultRenderer
 
   override def createStatementForQuery(options: Neo4jOptions): String = options.query.value
 
   override def createStatementForRelationships(options: Neo4jOptions): String = {
-    s"""MATCH (${Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS}:${options.relationshipMetadata.source.labels.mkString(":")})
-       |MATCH (${Neo4jUtil.RELATIONSHIP_TARGET_ALIAS}:${options.relationshipMetadata.target.labels.mkString(":")})
-       |MATCH (${Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS})-[${Neo4jUtil.RELATIONSHIP_ALIAS}:${options.relationshipMetadata.relationshipType}]->(${Neo4jUtil.RELATIONSHIP_TARGET_ALIAS})
-       |RETURN ${Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS}, ${Neo4jUtil.RELATIONSHIP_ALIAS}, ${Neo4jUtil.RELATIONSHIP_TARGET_ALIAS}
-       |""".stripMargin
+    val sourcePrimaryLabel = options.relationshipMetadata.source.labels.head
+    val sourceOtherLabels = options.relationshipMetadata.source.labels.takeRight(options.nodeMetadata.labels.size - 1)
```

Comment:
Scusa ma non dovrebbe essere così?
```suggestion
    val sourceOtherLabels = options.relationshipMetadata.source.labels.takeRight( options.relationshipMetadata.source.labels.size - 1)
```